R/metab.bookkeep.R

```diff
@@ -73,7 +73,7 @@ metab.bookkeep <- function(do.obs, do.sat, k.gas, z.mix, irr, ...){
 		warning("datetime not found, inferring sampling frequency from # of observations") # issue a warning (note checks in addNAs)
 		# NOTE: because of the checks in addNA's, it is unlikely a user would receive this warning via metab()
 		# warning will only be seen through direct use of metab.bookkeep when datettime is not supplied
-		freq <- nobs
+		freq <- nobs - 1
```

Comment:
It's all rate-based anyway. So you take the mean of the deltas, then
multiply it by what ever time period it should apply over. I know what you
mean about it "missing" a period in there, though. But realize that if that
it doesn't matter at all if that deltaDO is equal to the average of the
other nighttime deltaDO's (because we take the mean then multiply by freq).

On Wed, Sep 7, 2016 at 4:59 PM, Jake Zwart <notifications@github.com> wrote:

> In R/metab.bookkeep.R
> <https://github.com/GLEON/LakeMetabolizer/pull/121#discussion_r77902741>:
>
> > @@ -73,7 +73,7 @@ metab.bookkeep <- function(do.obs, do.sat, k.gas, z.mix, irr, ...){
> >  		warning("datetime not found, inferring sampling frequency from # of observations") # issue a warning (note checks in addNAs)
> >  		# NOTE: because of the checks in addNA's, it is unlikely a user would receive this warning via metab()
> >  		# warning will only be seen through direct use of metab.bookkeep when datettime is not supplied
> > -		freq <- nobs
> > +		freq <- nobs - 1
>
> I *have* switched to midnight to midnight side because if you apply a
> mass balance algebraic approach to O2 data (e.g. bookkeeping), then you
> will be missing a portion of the mass balance for O2 in a day using just a
> 1 midnight. Maybe less important for non-algebraic approaches.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GLEON/LakeMetabolizer/pull/121/files/c75567a84f7677ee3238146a630d9ff014c790da#r77902741>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFXrf5Q2iei5XcpzEmrusT2E0mswys2Kks5qnyXMgaJpZM4Jvljp>
> .
>



-- 
Ryan D. Batt
Aquatic Ecologist
Postdoctoral Researcher, Rutgers University
Website: http://limnology.wisc.edu/personnel/batt/